File: ipification-sdk-android-java/app/src/main/java/com/ipification/demoapp/activity/SuccessResultActivity.java
package com.ipification.demoapp.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;

import com.ipification.demoapp.R;
import com.ipification.demoapp.data.TokenInfo;
import com.ipification.demoapp.databinding.ActivityPhoneVerifyBinding;
import com.ipification.demoapp.databinding.ActivitySuccessResultBinding;

import org.json.JSONException;
import org.json.JSONObject;

public class SuccessResultActivity extends AppCompatActivity {

    private ActivitySuccessResultBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySuccessResultBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        initView();
        setupActionBar();
    }

    private void initView() {
        String tokenInfo = getIntent().getStringExtra("responseStr");
        JSONObject jsonObject = null;
        try {
            jsonObject = new JSONObject(tokenInfo != null ? tokenInfo : "");
            binding.tvMainDetail.setText(jsonObject.toString(4)); // 4 is the number of spaces for indentation

        } catch (JSONException e) {
            binding.tvMainDetail.setText(tokenInfo);
        }

    }

    private void setupActionBar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle("Result");
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        } else {
            return super.onOptionsItemSelected(item);
        }
    }
}

File: ipification-sdk-android-kotlin/app/src/main/java/com/ipification/demoapp/activity/SuccessResultActivity.kt
package com.ipification.demoapp.activity

import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.ipification.demoapp.data.TokenInfo
import com.ipification.demoapp.databinding.ActivitySuccessResultBinding

class SuccessResultActivity : AppCompatActivity() {
    lateinit var binding: ActivitySuccessResultBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySuccessResultBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setup()
    }

    private fun setup(){
        val tokenInfo = intent.getStringExtra("responseStr")
        binding.tvMainDetail.text = tokenInfo
        val actionbar = supportActionBar
        //set actionbar title
        actionbar!!.title = "Result"
        actionbar.setDisplayHomeAsUpEnabled(true)
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                // API 5+ solution
                onBackPressed()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}